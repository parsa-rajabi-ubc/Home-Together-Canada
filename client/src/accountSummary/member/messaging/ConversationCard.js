/**
 *
 * @Author:     Jeff Hatton
 * @Created:    2021.3.04
 *
 * @Description: Member Messaging Conversation Card
 *
 */
import React,{useState} from 'react';
import PropTypes from "prop-types";
import Moment from 'react-moment';

function ConversationCard(props) {
    const {userId, messageContent, datePosted,onClick} = props;

    const [userName,setUserName] = useState("");



    return (
        <section className={"card-container mt-0"} onClick={onClick}>
            <div className={"inline ml-6 align-middle "}>
                {/*need to be fix*/}
                <label className={"font-semibold justify-between"}> {userId}</label>

                <section className={"inline float-right pr-6 justify-end"}>
                    <Moment format="MMM D, YYYY">{datePosted}</Moment>
                </section>
            </div>
            <div className={"ml-6 mt-4 flex"}>
                <label className={"w-full font-light"}> {messageContent} </label>
            </div>
        </section>
    );
}

ConversationCard.propTypes = {
    userId: PropTypes.number.isRequired,
    messageContent: PropTypes.string.isRequired,
    datePosted: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired
}


export default ConversationCard;