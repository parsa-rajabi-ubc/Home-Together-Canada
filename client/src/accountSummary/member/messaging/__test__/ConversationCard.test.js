/**
 * @Author:     Jeff Hatton
 * @Created:    2021.3.06
 *
 * @Description: Member Listing Card Test
 *
 */

import React from 'react';
import ShallowRenderer from "react-test-renderer/shallow";
import ConversationCard from "../ConversationCard";

describe('ConversationCard', () => {
    describe('Snapshot test', () => {
        it('should match snapshot test', () => {
            //given
            const props = {
                messageContent: "message content 18 other hour",
                userName: "otherUser18",
                datePosted: "2021-01-14T01:42:39.005Z"
            }

            //when
            const renderer = new ShallowRenderer();
            renderer.render(<ConversationCard onClick={jest.fn()} {...props}/>);
            const result = renderer.getRenderOutput();
            //then
            expect(result).toMatchSnapshot();
        });
    });
});