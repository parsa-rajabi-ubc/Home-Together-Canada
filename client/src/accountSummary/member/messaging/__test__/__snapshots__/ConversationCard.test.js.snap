// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationCard Snapshot test should match snapshot test 1`] = `
<div>
  <section
    className="card-container mt-0"
    onClick={[MockFunction]}
  >
    <div
      className="inline ml-6 align-middle "
    >
      <label
        className="font-semibold justify-between"
      >
         
        otherUser18
      </label>
      <section
        className="inline float-right pr-6 justify-end"
      >
        <n
          ago={false}
          calendar={false}
          decimal={false}
          element={null}
          filter={[Function]}
          format="MMM D, YYYY"
          fromNow={false}
          interval={60000}
          local={false}
          onChange={[Function]}
          titleFormat=""
          toNow={false}
          trim={false}
          unit={null}
          unix={false}
          utc={false}
          withTitle={false}
        >
          2021-01-14T01:42:39.005Z
        </n>
      </section>
    </div>
    <div
      className="ml-6 mt-4 flex"
    >
      <label
        className="w-full font-light"
      >
         
        message content 18 other hour
         
      </label>
    </div>
  </section>
</div>
`;
